### PR TITLE
[json-rpc] ensure response data use lower case for hex-encoded string

### PR DESCRIPTION
**Pull request by xli**
## Motivation

JSON-RPC api response for account address case format is inconsistent: some uses lower case, some uses upper case.

Consistent lower case format can avoid people make mistake like: string compare without ignoring case.
Especially for account address, instead of comparing bytes, developers may direct compare hex-encoded string, which should be case insensitive.
This PR also changes other hex-encoded data type to use BytesView for ensuring lower case across board.

## Test Plan

unit tests

## Related PRs

#6296

### CODE REVIEW

**Comment by libra-action** [comment]
**This PR may have modified JSON-RPC server code.**

Breaking changes policy:

1. Chaning JSON-RPC method signatures or removing/modifying fields
   in the response in /V1 will not be accepted.
2. Adding fields is ok.
3. Adding a new JSON-RPC method is ok.

Please ensure you have also done the following to update the docs:

1. Update json-rpc/API-CHANGELOG.md.
2. Add/update type/method documentation under json-rpc/docs.

File: json-rpc/src/methods.rs [hunk 1]
```diff
@@ -534,7 +536,7 @@ async fn get_account_transactions(
 
         all_txs.push(TransactionView {
             version: tx.version,
-            hash: tx.transaction.hash().to_hex(),
+            hash: BytesView::from(tx.transaction.hash().to_vec()),
             bytes: lcs::to_bytes(&tx.transaction)?.into(),
```
Comment (thefallentree) [inline json-rpc/src/methods.rs:540]: see here,  if we declare everything in response field as bytesview directly we could sae all these BytesView::from()

**Review by thefallentree — APPROVED** [review]
Great change! Thanks for making the codebase more consistent and clean!

**Comment by xli** [comment]
\land

**Pull request closed without merging**